Code/Framework/AzFramework/AzFramework/Physics/Material.cpp

```diff
@@ -47,7 +47,26 @@ namespace Physics
             {
                 materialSelection->SetMaterialSlots(Physics::MaterialSelection::SlotsArray());
             }
-            materialSelection->SyncSelectionToMaterialLibrary();
+
+            // Not validating the material selection against the library at asset builder time
+            // because the library is not available. This happens when a MaterialSelection
+            // is part of a prefab. This will avoid writing null material ids into prefabs.
+            if (!IsAssetBuilder())
+            {
+                materialSelection->SyncSelectionToMaterialLibrary();
+            }
+        }
+
+        bool IsAssetBuilder() const
+        {
+            AZ::SerializeContext* serializeContext = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
+
+            // TypeId of AssetBuilderComponent
+            const char* typeIDAssetBuilderComponent = "{04332899-5d73-4d41-86b7-b1017d349673}";
+
+            return serializeContext
+                && serializeContext->FindClassData(AZ::TypeId::CreateString(typeIDAssetBuilderComponent));
```

Comment:
You could just query for the SettingsRegistry for the assetbuilder specialization.
Here is how you can do it: https://github.com/o3de/o3de/pull/8800#discussion_r846351409

Reply:
Thanks @lumberyard-employee-dm for the suggestion. I'll pass on this change for now since I've already tested and verified the current solution works in all scenarios and PhysX Material Library is going to go away soon anyway (so this code will be removed). I'll keep your method in mind if I have to do a similar operation again in the future. Cheers.